src/extension/infer.rs

```diff
@@ -1133,6 +1160,502 @@ mod test {
+        let abc = ExtensionSet::from_iter([A, B, C]);
+        let bc = ExtensionSet::from_iter([B, C]);
+        let b = ExtensionSet::singleton(&B);
+        let c = ExtensionSet::singleton(&C);
+
+        let mut hugr = Hugr::new(NodeType::open_extensions(ops::CFG {
+            signature: FunctionType::new(type_row![NAT], type_row![NAT]).with_extension_delta(&abc),
+        }));
+
+        let root = hugr.root();
+
+        let [entry, entry_in, entry_out] = create_with_io(
+            &mut hugr,
+            root,
+            ops::BasicBlock::DFB {
+                inputs: type_row![NAT],
+                other_outputs: type_row![],
+                predicate_variants: vec![type_row![NAT], type_row![NAT]],
+                extension_delta: a.clone(),
+            },
+            FunctionType::new(vec![NAT], twoway(NAT)),
+        )?;
+
+        let exit = hugr.add_node_with_parent(
+            hugr.root(),
+            NodeType::open_extensions(ops::BasicBlock::Exit {
+                cfg_outputs: type_row![NAT],
+            }),
+        )?;
+
+        let mkpred = hugr.add_node_with_parent(
+            entry,
+            NodeType::open_extensions(make_opaque(
+                A,
+                FunctionType::new(vec![NAT], twoway(NAT)).with_extension_delta(&a),
+            )),
+        )?;
+
+        let bb0 = make_block(
+            &mut hugr,
+            root,
+            ops::BasicBlock::DFB {
+                inputs: type_row![NAT],
+                other_outputs: type_row![],
+                predicate_variants: vec![type_row![NAT]],
+                extension_delta: bc.clone(),
+            },
+            FunctionType::new(vec![NAT], oneway(NAT)),
+            make_opaque(
+                B,
+                FunctionType::new(vec![NAT], oneway(NAT)).with_extension_delta(&bc),
+            ),
+        )?;
+
+        let bb1 = make_block(
+            &mut hugr,
+            root,
+            ops::BasicBlock::DFB {
+                inputs: type_row![NAT],
+                other_outputs: type_row![],
+                predicate_variants: vec![type_row![NAT], type_row![NAT]],
+                extension_delta: b.clone(),
+            },
+            FunctionType::new(vec![NAT], twoway(NAT)),
+            make_opaque(
+                B,
+                FunctionType::new(vec![NAT], twoway(NAT)).with_extension_delta(&b),
+            ),
+        )?;
+
+        let bb10 = make_block(
+            &mut hugr,
+            root,
+            ops::BasicBlock::DFB {
+                inputs: type_row![NAT],
+                other_outputs: type_row![],
+                predicate_variants: vec![type_row![NAT]],
+                extension_delta: c.clone(),
+            },
+            FunctionType::new(vec![NAT], oneway(NAT)),
+            make_opaque(
+                C,
+                FunctionType::new(vec![NAT], oneway(NAT)).with_extension_delta(&c),
+            ),
+        )?;
+
+        let bb11 = make_block(
```

Comment:
```suggestion
        let bb11 = bb10.clone();
```

Reply:
Not quite, because `make_block` is adding a bunch of nodes to the graph, but I can certainly abstract this a bit!

Reply:
ah yes, good point, `clone`ing an op doesn't clone its children, sorry!